### DISCUSSION TITLE
Screen aware shortcuts

### DISCUSSION THREAD

**Discussion by Orphis** [rxhanson/Rectangle#814] [Ideas]
It would be great if Rectangle was aware of which screen it was on and could switch to a different set of shortcuts accordingly.

A scenario I'd like to have supported is that when I switch from the home office with an ultrawide screen (32/9), I prefer my shortcuts to do first quarter / middle half / last quarter, but when I'm using another regular screen (16/9), I prefer to have the shortcuts do first half / fullscreen / second half.

A way to implement this could be to lookup on which screen a window is before it's moved and use a different action associated with a specific screen, or maybe a specific aspect ratio.

Alternatively, if there was an API I could use to load settings for Rectangle, I could externalize it (detect the screen setup and load the right config).

**Comment by rxhanson** [comment]
I like this idea, but don't have the bandwidth to add it in at this time. I see no reason to turn down a good PR for it.

**Comment by joeyhoer** [comment]
I have been trying to accomplish the same thing. I was planning to use the following AppleScript to detect the connected monitors by name, and then perform a command line action to load in the settings. Unfortunately (as far as I am aware) there isn't a command to dynamically change the configuration of Rectangle. I considered a work around of quitting Rectangle, moving the appropriate config file to `~/Library/Application Support/Rectangle/RectangleConfig.json` and then reopening the app. Theoretically that should work, but it would require the script to be run manually, or polling to allow the script to be run when a monitor is connected/disconnected—which isn't the most elegant solution.

```applescript

tell application "System Events"
	set {display_names} to paragraphs of (do shell script "system_profiler SPDisplaysDataType | awk '/^ {8}[^ ]+/{sub(/^ {8}/,\"\",$0); sub(/:$/,\"\",$0); display_name=$0; printf \"%s\\n\", display_name}'")
	
	if display_names contains "LS49A950U" then
		# Load ultrawide config
		do shell script "cp \"$HOME/Library/Application Support/Rectangle/LS49A950U_RectangleConfig.json\" \"$HOME/Library/Application Support/Rectangle/RectangleConfig.json\""
	else
		# Load standard config
		do shell script "cp \"$HOME/Library/Application Support/Rectangle/LS49A950U_RectangleConfig.json\" \"$HOME/Library/Application Support/Rectangle/RectangleConfig.json\""
		quit application "Rectangle"
		repeat while (exists process "Rectangle")
			delay 0.1
		end repeat
		launch application "Rectangle"
	end if
	
	# Relaunch Rectangle
	quit application "Rectangle"
	repeat while (exists process "Rectangle")
		delay 0.1
	end repeat
	launch application "Rectangle"
end tell
```

**Reply by rxhanson** [reply]
Thanks for sharing your script. All of this would definitely better handled in the app, in Swift, rather than AppleScript, but I still don't have the bandwidth for it. A minimal solution of loading a config via command line wouldn't be that hard to add in, if anyone wants to give it a go. You can check out the existing URL parameters logic for programmatically executing Rectangle actions via URL, and build out something where it takes a config as a parameter and just executes the load config functionality. You can also easily set up detection of displays here, too, as a lot of that type of thing is already in the app as well.
